xarray/core/dataset.py

```diff
@@ -6651,6 +6683,20 @@ def pad(
                     end_values=end_values,
                     reflect_type=reflect_type,
                 )
+            elif name in var_pad_width.keys() and not isinstance(
+                pad_width[name], int
+            ):  # dimension coordinates
+                w0, w1 = pad_width[name]  # type: ignore
+                fill_value_ind = dtypes.get_fill_value(var.dtype)
+                if isinstance(w0, int):
+                    w0_ = IndexVariable(name, [fill_value_ind] * w0)
+                else:
+                    w0_ = IndexVariable(name, w0)
+                if isinstance(w1, int):
+                    w1_ = IndexVariable(name, [fill_value_ind] * w1)
+                else:
+                    w1_ = IndexVariable(name, w1)
+                variables[name] = var.concat([w0_, var, w1_], dim=name)
```

Comment:
I'm a bit wondering if we put this series of logic should be moved into `IndexVariable.pad`.

Reply:
Yes I think that would make sense.

Reply:
Thanks. Done.